plugins/woocommerce-blocks/assets/js/base/context/hooks/collections/use-collection-data.ts

```diff
@@ -155,7 +156,7 @@ export const useCollectionData = < T >( {
 		return buildCollectionDataQuery( collectionDataQueryState );
 	}, [ collectionDataQueryState ] );
 
-	return useCollection< T >( {
+	return useCollection< WCStoreV1ProductsCollectionProps, false >( {
```

Comment:
> `false` is the default value for the second parameter, so what is the benefit of passing it? For explicitly?

That's correct. I guess we can get rid of it now.

> * Because the only place the `useCollection` hook doesn't return the array is inside `useCollectionData`, I think we should restore the original type of `useCollection` and cast its return type in `useCollectionData`.

After diving into the code, you're right. I agree the best solution is restoring the type handling in the useCollection hook.